sdk/core/core-lro/src/lroEngine/impl.ts

```diff
@@ -0,0 +1,336 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  GetLroStatusFromResponse,
+  LongRunningOperation,
+  LroBody,
+  LroInfo,
+  LroResourceLocationConfig,
+  LroResponse,
+  LroStatus,
+  PollerConfig,
+  RawResponse,
+  ResumablePollOperationState,
+} from "./models";
+import { PollOperationState } from "../pollOperation";
+import { logger } from "./logger";
+
+export function throwIfUndefined<T>(
+  input: T | undefined,
+  options: { errorMessage?: string } = {}
+): T {
+  if (!input) {
+    throw new Error(options.errorMessage ?? "undefined variable");
```

Comment:
This is checking `falsy` not undefined, `""` would fall in here and the error message would be missleading

Reply:
You're right, I updated the check to be for undefined instead.